### PR TITLE
[Feature] Layer-wise Prefill

### PR DESCRIPTION
**Pull request by jason-fxz**
This PR implements layer-wise prefill functionality by adding a new `forward_split_prefill` method to all supported model architectures. This allows launching several decoder layers instead of processing the entire model at once, with intermediate states stored in the forward batch.

We will initially support dense text models. Supporting MoE models or multimodal models needs more investigation.

## Motivation

This feature is essential for [PDMultiplexing](https://arxiv.org/abs/2504.14489) implementation.

## Modifications

1. **Added new ForwardMode**: `SPLIT_PREFILL` to `ForwardMode` enum with corresponding helper method `is_split_prefill()`

2. **Extended ForwardBatch class** with split prefill specific fields:
   - `hidden_states`: Intermediate hidden states storage
   - `residual`: Intermediate Residual connection states  
   - `model_specific_states`: Model-specific intermediate states
   - `split_index`: Current layer index being processed

3. **Extended ModelRunner** with `forward_split_prefill()` method.
4. **Models** Added `forward_split_prefill()` method for supported models:
    - [x] Qwen
    - [x] Llama
    - [x] Gemma
    - [ ] ...

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Review by hnyls2002 — APPROVED** [review]

**Comment by hnyls2002** [comment]
@jason-fxz Please fix this https://github.com/sgl-project/sglang/actions/runs/16249493498/job/45951257467?pr=7634#step:4:4034

**Comment by jason-fxz** [comment]
@hnyls2002 Fixed by [add split_index to output_dict](https://github.com/sgl-project/sglang/pull/7634/commits/652298f0a7b8c25efa24b494eed72f3d01512d79)

**Comment by merrymercy** [comment]
Can you also include a test case?

**Comment by Zhangmj0621** [comment]
@jason-fxz  Hello, do your pr support pd disaggregation?

**Comment by jason-fxz** [comment]
> @jason-fxz Hello, do your pr support pd disaggregation?

This PR just provides the `forward_split_prefill()` interface. When ModelRunner performs a forward pass and the `ForwardMode` is set to `SPLIT_PREFILL`, it will call `forward_split_prefill()` and store intermediate states in `ForwardBatch`.

`forward_split_prefill()` will return `None` until the final transformer layer is reached. At the last layer, it returns the `next_token_logits` as usual.

**Comment by jason-fxz** [comment]
> Can you also include a test case?

Done by [PR8231](https://github.com/sgl-project/sglang/pull/8231)
